### PR TITLE
[Doc] Document DLS limitation for aggregation

### PR DESCRIPTION
**Pull request by ywangd**
This PR clarifies the limitation for DLS in terms of aggreation.

### CODE REVIEW

**Review by jpountz — APPROVED** [review]
This looks accurate to me!

File: x-pack/docs/en/security/limitations.asciidoc [hunk 1]
```diff
@@ -79,9 +79,15 @@ including the following queries:
...
 * A search request cannot be profiled if document level security is enabled.
-* The <<search-terms-enum,terms enum API>> does not return terms if document 
+* The <<search-terms-enum,terms enum API>> does not return terms if document
 level security is enabled.
 
+While document-level security prevents users from seeing documents that they do not have access to,
+it is possible to craft search requests in order to discover aggregate information about the entire
+index. A user whose access to an index is restricted via document-level security could still learn
+about fields names and terms that only exist in inaccessible documents and count how many inaccessible
+documents containing a given term. But no more than that.
```
Comment (lockewritesdocs) [inline x-pack/docs/en/security/limitations.asciidoc:89]:
```suggestion
NOTE: While document-level security prevents users from viewing restricted documents,
it's still possible to write search requests that return aggregate information about the
entire index. A user whose access is restricted to specific documents in an index could
still learn about field names and terms that only exist in inaccessible
documents, and count how many inaccessible documents contain a given term. 
```

Comment (lockewritesdocs) [reply x-pack/docs/en/security/limitations.asciidoc:89]: @ywangd, are there security implications here? If a user can count how many inaccessible documents can contain specific field names and terms, can they do anything? I'm just curious how important it is to tell users about this limitation.

**Review by lockewritesdocs — COMMENTED** [review]
I left a suggested change and also a question about how pertinent this limitation is. Thanks @ywangd 👍

**Comment by ywangd** [comment]
> are there security implications here? If a user can count how many inaccessible documents can contain specific field names and terms, can they do anything?

That's literally what they can do in the context of Elasticsearch. Whether this information is useful enough for them to derive other information in some other context is at users' discretion. But it is important for us to let them know about this limitation so they are able to make informed assessment.

**Comment by ywangd** [comment]
@elasticmachine update branch

**Review by lockewritesdocs — APPROVED** [review]
LGTM 🦖

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/docs

**Pull request merged**